edx_proctoring/api.py

```diff
@@ -1526,6 +1526,23 @@ def _calculate_allowed_mins(due_datetime, allowed_mins):
     return actual_allowed_mins, is_exam_past_due_date
 
 
+def _is_self_paced_course(user_id, course_id):
```

Comment:
@schenedx Thanks for the info.

@afzaledx @ibrahimahmed443 Can you figure out if there is something going wrong with processing the course publish event when switching between self-paced/instructor-paced? If so, let's fix it. If we can't rely on the due_date on the exam object we should be able to use the due_date from the context instead.